### PR TITLE
Prepare the Siemply POC

### PR DESCRIPTION
**Pull request by Siemienik**
The first POC of Siemply handlers library for AWS-lambda

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,13 @@
 {
-  "name": "Siemply",
+  "name": "@siemply/aws-lambda",
   "version": "0.0.0-wip1",
```
Comment (Siemienik) [inline package.json:3]:
```suggestion
  "version": "1.0.0-alpha1",
```

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+// TODO install and configure eslinter, add GH action for auto checking
+// TODO cover with unit tests. (jest?)
+// TODO extract to non `index.ts` file.
+// TODO uml with desired architecture
+// TODO docs
+export * from './core';
```
Comment (Siemienik) [inline src/index.ts:7]:
```suggestion
// TODO aws-cdk utils like auto mapper handlers to restful api gateways
export * from './core';
```

File: src/http.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    HttpServerErrorResponses
+    ;
+
+/**
+ * @TODO convert to dir, reexport, for instance:
+ * /src/html
+ *        - statusCodes/
+ *              2xxSuccessfulResponses.ts
+ *              5xxInternalServerError.ts
+ */
```
Comment (Siemienik) [inline src/http.ts:35]:
```suggestion
 */
 
```

File: src/errors.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * @description HTTP code 404: Not Found
+ */
+class HttpNotFoundError extends HttpError {
+    constructor(debug = false) {
+        super(STATUS_404_NOT_FOUND, "Not Found.", debug);
+    }
+}
+
+
+  // todo fullfill http errors
```
Comment (Siemienik) [inline src/errors.ts:34]:
```suggestion
  // todo fullfill http errors
  
```

File: src/params.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      throw new HttpBadRequestError();
+    }
+    return $event.pathParameters[name];
+  };
+
+  // TODO add flag optional, which suppress throwing an error
+  // TODO add mapper for value to map from string to any another 
+  // TODO add default value
+  // TODO add fromQuery, fromBody etc.
+
```
Comment (Siemienik) [inline src/params.ts:25]:
```suggestion

```

File: src/examples.ts [hunk 6]
```diff
@@ -0,0 +1,282 @@
+import type {
+    APIGatewayProxyEventV2WithLambdaAuthorizer,
+    APIGatewayProxyResultV2
+} from "aws-lambda";
+
```
Comment (Siemienik) [inline src/examples.ts:5]:
```suggestion

// TODO exclude this file from final package
```

File: package.json [hunk 7]
```diff
@@ -1,12 +1,13 @@
...
-  "version": "0.0.0-wip1",
+  "name": "@siemply/aws-lambda",
+  "version": "1.0.0-alpha1",
   "description": "Siemply is super SIMPLE handlers framework for AWS Lambda. ",
   "main": "lib/index.js",
   "repository": "https://github.com/Siemienik/Siemply",
   "author": "Siemienik Pawel <hi@siemienik.com>",
   "license": "MIT",
   "devDependencies": {
+    "@types//aws-lambda": "^8.10.106",
```
Comment (Siemienik) [inline package.json:10]:
```suggestion
    "@types/aws-lambda": "^8.10.106",
```

File: src/errors.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ */
+export class HttpNotFoundError extends HttpError {
+    constructor(debug = false) {
+        super(STATUS_404_NOT_FOUND, "Not Found.", debug);
+    }
+}
+
+
+  // todo fullfill http errors
+
```
Comment (Siemienik) [inline src/errors.ts:35]:
```suggestion

```

File: src/http.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
```
Comment (Siemienik) [inline src/http.ts:36]:
```suggestion

```

File: src/params.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+      throw new HttpBadRequestError();
+    }
+
+    return value;
+  };
+
+  // TODO add flag optional, which suppress throwing an error
+  // TODO add mapper for value to map from string to any another 
+  // TODO add default value
+  // TODO add fromQuery, fromBody etc.
```
Comment (Siemienik) [inline src/params.ts:28]:
```suggestion
  // TODO add fromQuery, fromBody etc.

```

File: src/core.ts [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+        try {
+            const { $statusCode, $result, $event, ...result } = steps.reduce(
+                (payload, step) => ({ ...payload, ...step(payload) }),
+                { $event: event }
+            );
+
+            return {
+                statusCode: $statusCode || STATUS_200_OK,
+                body: $result || result[Object.keys(result).pop() as keyof typeof result],
+                // TODO headers, cookies
```
Comment (Siemienik) [inline src/core.ts:32]:
```suggestion
                // TODO headers, cookies. Followed by: @see https://github.com/Siemienik/Siemply/issues/3
```

File: src/errors.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+ * @description HTTP code 404: Not Found
+ */
+export class HttpNotFoundError extends HttpError {
+    constructor(debug = false) {
+        super(STATUS_404_NOT_FOUND, "Not Found.", debug);
+    }
+}
+
+
+  // todo fullfill http errors
```
Comment (Siemienik) [inline src/errors.ts:34]:
```suggestion
  // todo fullfill http errors. Followed by: @see https://github.com/Siemienik/Siemply/issues/6

```

File: src/core.ts [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+            };
+
+        } catch (err) {
+            const {
+                debug = true,
+                statusCode = STATUS_500_INTERNAL_SERVER_ERROR,
+                publicMessage: body = "Internal Server Error." // TODO consider about persist as consts the http body messages for errors. 
+            } = err as HttpError; // TODO redesign: It could be just a Error - that case is fine beacuse default values. However it could be not an Object (edge-case: err is a string), then app will crush.
```
Comment (Siemienik) [inline src/core.ts:40]:
```suggestion
            } = err as HttpError; // TODO redesign: It is not always HttpError. It could be just an Error, for that case is not too bad beacuse then the default values will be provided. However, in JS non Object variable could be thrown (it's edge-case for instance err could be a string).
```

File: src/index.ts [hunk 14]
```diff
@@ -0,0 +1,8 @@
+
+// TODO install and configure eslinter, add GH action for auto checking
+// TODO cover with unit tests. (jest?)
```
Comment (Siemienik) [inline src/index.ts:3]:
```suggestion
// TODO cover with unit tests. (jest?). Followed by: @see https://github.com/Siemienik/Siemply/issues/5
```

File: src/http.ts [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+// TODO create & fullfill 3xx status codes
+// TODO fullfill 4xx status codes
+export const STATUS_400_BAD_REQUEST = 400;
+export const STATUS_404_NOT_FOUND = 404;
+export type HttpClientErrorResponses =
+    typeof STATUS_400_BAD_REQUEST |
+    typeof STATUS_404_NOT_FOUND
+    ;
+
+// TODO fullfill 5xx status codes
```
Comment (Siemienik) [inline src/http.ts:17]:
```suggestion
// TODO create & fullfill 1xx status codes. Followed by: @see https://github.com/Siemienik/Siemply/issues/4
// TODO fullfill 2xx status codes. Followed by: @see https://github.com/Siemienik/Siemply/issues/4
export const STATUS_200_OK = 200;
export type HttpSuccessfulResponses =
    typeof STATUS_200_OK
    ;

// TODO create & fullfill 3xx status codes. Followed by: @see https://github.com/Siemienik/Siemply/issues/4
// TODO fullfill 4xx status codes. Followed by: @see https://github.com/Siemienik/Siemply/issues/4
export const STATUS_400_BAD_REQUEST = 400;
export const STATUS_404_NOT_FOUND = 404;
export type HttpClientErrorResponses =
    typeof STATUS_400_BAD_REQUEST |
    typeof STATUS_404_NOT_FOUND
    ;

// TODO fullfill 5xx status codes. Followed by: @see https://github.com/Siemienik/Siemply/issues/4
```

File: src/index.ts [hunk 16]
```diff
@@ -0,0 +1,8 @@
+
+// TODO install and configure eslinter, add GH action for auto checking
```
Comment (Siemienik) [inline src/index.ts:2]:
```suggestion
// TODO install and configure eslinter, add GH action for auto checking. Followed by: https://github.com/Siemienik/Siemply/issues/7
```

**Review by Siemienik — COMMENTED** [review]
Added links to issues
